core/authority-discovery/src/lib.rs

```diff
@@ -112,23 +123,31 @@ where
 	pub fn new(
 		client: Arc<Client>,
 		network: Arc<Network>,
-		dht_event_rx: Receiver<DhtEvent>,
+		key_store: BareCryptoStorePtr,
+		dht_event_rx: Pin<Box<dyn Stream<Item = DhtEvent> + Send>>,
 	) -> Self {
 		// Kademlia's default time-to-live for Dht records is 36h, republishing records every 24h. Given that a node
```

Comment:
Not perfect to depend on any "default value" that probably could change at any moment.

Reply:
Right. Kademlia does not expose these values to the outside as of today (https://github.com/libp2p/rust-libp2p/blob/master/protocols/kad/src/lib.rs). If we want to do that in the future I would prefer doing that in a follow up pull request.